### PR TITLE
web: show token

### LINKED ISSUE #95: web: show token

### PR DESCRIPTION
**Pull request by SveaGaenzle**
I have tried to display the token on the web.

## Issue:

- resolves #95 

## Description:
- I have added `token` to the `User`-interface
- I have added to the `user dashboard` in the `loader` that the user's token and username are returned. And I changed the `UserDashboard` so that the token is displayed. If no token exists, _"no token exists"_ is displayed.

## Update
- I have changed in `addUser` that a random token is generated for new users created via the website
- I use uuid for the tokens.

### CODE REVIEW

File: comprl-web/app/db/sqlite.data.tsx [hunk 1]
```diff
@@ -17,17 +18,27 @@ db.close();
...
+    return { id: res.user_id, name: res.username, role: res.role } as User;
+}
+
+export async function getToken(username: string) {
+    const db = new Database('users.db', { verbose: console.log });
+    const stmt = db.prepare('SELECT * FROM users WHERE username = ?');
+    const res = stmt.get(username);
+    db.close();
+    if (!res) { return undefined }
+    return res.token;
```
Comment (Raining-Cloud) [inline comprl-web/app/db/sqlite.data.tsx:43]: I guess this should be in the getUser() function, like the User type gets another field named token.

File: comprl-web/app/routes/_dashboard.usr.$name.tsx [hunk 2]
```diff
@@ -23,12 +25,21 @@ export async function loader({ request, params }: LoaderFunctionArgs) {
...
 }
 
 export default function UserDashboard() {
-
+  const { token } = useLoaderData<typeof loader>();
   return (
-    <Typography variant="h1">User</Typography>
+    <>
+      <Typography variant="h1">User</Typography>
+      <Typography variant="body1">Token: {token}</Typography>
```
Comment (Raining-Cloud) [inline comprl-web/app/routes/_dashboard.usr.$name.tsx:42]: Ye well this is not really the way we can work with it due to web is a bit more complex than adding a headining ... :(
Also the <></> is  pretty sure not the way to go

**Review by Raining-Cloud — DISMISSED** [review]

File: comprl-web/app/db/sqlite.data.tsx (hunk 1 above)
Comment (SveaGaenzle) [reply comprl-web/app/db/sqlite.data.tsx]: Yes, you're right, that's better. 
I have changed this.

File: comprl-web/app/routes/_dashboard.usr.$name.tsx (hunk 2 above)
Comment (SveaGaenzle) [reply comprl-web/app/routes/_dashboard.usr.$name.tsx:40]: I don't quite understand what you mean by that. How should the token be displayed?

Comment (SveaGaenzle) [reply comprl-web/app/routes/_dashboard.usr.$name.tsx]: I changed it

**Review by Cari1111 — APPROVED** [review]
Thanks for the implementation. The Token is correctly generated and displayed.

**Pull request merged**